packages/gensx/src/commands/run.tsx

```diff
@@ -12,21 +12,102 @@ import { useProjectName } from "../hooks/useProjectName.js";
 import { getAuth } from "../utils/config.js";
 import { USER_AGENT } from "../utils/user-agent.js";
 
+// JSON-serializable value type for progress data
+export type JsonValue =
+  | string
+  | number
+  | boolean
+  | null
+  | JsonValue[]
+  | { [key: string]: JsonValue };
+
+// Individual message types
+export interface WorkflowStartMessage {
+  type: "start";
+  workflowExecutionId?: string;
+  workflowName: string;
+}
+
+export interface WorkflowComponentStartMessage {
+  type: "component-start";
+  componentName: string;
+  label?: string;
+  componentId: string;
+}
+
+export interface WorkflowComponentEndMessage {
+  type: "component-end";
+  componentName: string;
+  label?: string;
+  componentId: string;
+}
+
+export interface WorkflowDataMessage {
+  type: "data";
+  data: JsonValue;
+}
+
+export interface WorkflowEventMessage {
+  type: "event";
+  data: Record<string, JsonValue>;
+  label: string;
+}
+
+export interface WorkflowObjectMessage {
+  type: "object";
+  data: Record<string, JsonValue>;
```

Comment:
In @danfhernandez's latest PR, he's changing the type here to just be `JsonValue` so we'll probably need to sequence these PRs

https://github.com/gensx-inc/gensx/pull/791

Reply:
I guess its time to start sharing these types...